lib/ui/health/Covid19TestLocations.dart

```diff
@@ -479,16 +478,16 @@ class _TestLocation extends StatelessWidget{
   }
 
   String _getPeriodText(HealthLocationDayOfOperation period, LinkedHashMap<int,HealthLocationDayOfOperation> workingPeriods){
-  String openText = Localization().getStringEx("panel.covid19_test_locations.work_time.open_until","Open until");
-  String closedText = Localization().getStringEx("panel.covid19_test_locations.work_time.closed_until","Closed until");
-  if(_determineIsOpen(period)){ //This is the active Period
```

Comment:
I believe this line was the original cause of the bug. `_determineIsOpen` only really worked if `period` was the current day's period, however at this point it could be the period for any day. The new code checks that condition.